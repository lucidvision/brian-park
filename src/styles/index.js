export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
