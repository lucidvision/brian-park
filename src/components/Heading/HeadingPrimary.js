import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors, fontSizes } from 'styles'

HeadingPrimary.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  text: PropTypes.string.isRequired
}

HeadingPrimary.defaultProps = {
  color: colors.primary,
  size: fontSizes.large,
  text: ''
}

const StyledHeading = styled.h1`
  color: ${props => props.color};
  font-size: ${props => props.size};
  font-weight: 400;
  text-transform: uppercase;
  letter-spacing: 1.5rem;
`

export default function HeadingPrimary (props) {
  const { text, ...rest } = props
  return (
    <StyledHeading {...rest}>{props.text}</StyledHeading>
  )
}
