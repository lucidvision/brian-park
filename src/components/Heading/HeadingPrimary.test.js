import React from 'react'
import { mount } from 'enzyme'
import HeadingPrimary from './HeadingPrimary'
import { colors, fontSizes } from 'styles'

const createHeadingPrimary = (override) => {
  const props = {
    text: 'Unicorns',
    ...override
  }

  return mount(<HeadingPrimary {...props} />)
}

describe('HeadingPrimary', () => {
  const headingPrimary = createHeadingPrimary()

  it('renders correctly', () => {
    expect(headingPrimary).toMatchSnapshot()
  })

  it('renders with the correct props', () => {
    expect(headingPrimary.find('HeadingPrimary').props()).toEqual({
      color: colors.primary,
      size: fontSizes.large,
      text: 'Unicorns'
    })
  })

  it('renders an h1 with the correct text', () => {
    expect(headingPrimary.find('h1').props().children).toBe('Unicorns')
  })
})
