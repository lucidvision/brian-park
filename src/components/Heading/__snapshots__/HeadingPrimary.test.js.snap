// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingPrimary renders correctly 1`] = `
<HeadingPrimary
  color="rgba(0, 13, 85, 0.8)"
  size="6rem"
  text="Unicorns"
>
  <styled.h1
    color="rgba(0, 13, 85, 0.8)"
    size="6rem"
  >
    <h1
      className="sc-bdVaJa bykBkz"
      color="rgba(0, 13, 85, 0.8)"
      size="6rem"
    >
      Unicorns
    </h1>
  </styled.h1>
</HeadingPrimary>
`;
